src/states/DefenseState.js

```diff
@@ -475,16 +465,18 @@ class DefenseState extends React.Component {
             draw = removeRandomItem(tilePool);
             players[0].hand[draw]++;
 
-            this.timer = setTimeout(
-                () => {
-                    this.onTileClicked({target:{name:this.state.lastDraw}});
-                    this.setState({
-                        currentBonus: 0
-                    });
-                },
-                (this.state.settings.time + this.state.currentBonus) * 1000
-            );
-            this.timerUpdate = setInterval(this.updateTime, 100);
+            if (this.state.settings.useTimer) {
```

Comment:
This was the main issue, missing protecting re-initializing the timer only when it's selected.